### PR TITLE
Ac 16 customer will see a popup on adding an item

### PR DESCRIPTION
**Pull request by Madhavareddymukku**

### CODE REVIEW

File: ac_posystem/src/pages/Calculations.js [hunk 1]
```diff
@@ -0,0 +1,34 @@
+import React from 'react'
+
+class Calculations extends React.Component {
```
Comment (wazmaaali) [inline ac_posystem/src/pages/Calculations.js:3]: Where are you using this class? If its not being used anywhere then delete this.

File: ac_posystem/src/pages/Home.js [hunk 2]
```diff
@@ -15,40 +22,57 @@ const Home = () => {
...
-        console.log("Error: ", err);
+        console.log("99999 Error: ", err);
       }
     };
     fetchCategories();
   }, []);
 
+  const newLocal = "iconLink";
   return (
+    // <div dangerouslySetInnerHTML={ {__html: htmlContent} } />
```
Comment (wazmaaali) [inline ac_posystem/src/pages/Home.js:39]: remove unnecessary line.

File: ac_posystem/src/pages/MainCategory.js [hunk 3]
```diff
@@ -129,244 +93,184 @@ const MainCategory = () => {
...
-  var item = addTOCart.find((x) => x.sc_id == datalist[index].sc_id);
-  if (item) {
-    item.count = item.count + 1;
-  } else {
-    var item = JSON.parse(JSON.stringify(datalist[index])); //just to make sure its not passing reference
-    item.count = 1;
-    addTOCart.push(item);
+let refreshPage = (e) => {
+  var val = e.target.value;
+  if (val == "Vegetables") {
```
Comment (wazmaaali) [inline ac_posystem/src/pages/MainCategory.js:334]: Vegetables,Fruits etc could have been declare once as enum and then used

File: ac_posystem/src/pages/MainCategory.js [hunk 4]
```diff
@@ -129,244 +93,184 @@ const MainCategory = () => {
...
-  console.log("INDEX == ", index);
-  var item = addTOCart.find((x) => x.sc_id == datalist[index].sc_id);
-  if (item) {
-    item.count = item.count + 1;
-  } else {
-    var item = JSON.parse(JSON.stringify(datalist[index])); //just to make sure its not passing reference
-    item.count = 1;
-    addTOCart.push(item);
+let refreshPage = (e) => {
+  var val = e.target.value;
```
Comment (wazmaaali) [inline ac_posystem/src/pages/MainCategory.js:333]: Use loop instead of so many if else commands.

File: ac_posystem/src/pages/MainCategory.js [hunk 5]
```diff
@@ -129,244 +93,184 @@ const MainCategory = () => {
...
-let setTitle = (id) => {
-  if (id == 1) {
-    document.getElementById("maintitle").innerHTML = "VEGETABLES";
+const handleClick = (e) => {
+  console.log(e.target.id, "999 : ", datalist[1]);
+  const a = e.target.id;
+  if (a == "b1") {
+    addTOCart.push(datalist[0]);
   }
-  if (id == 2) {
```
Comment (wazmaaali) [inline ac_posystem/src/pages/MainCategory.js:305]: Use loop instead of if else.

File: ac_posystem/src/pages/MainCategory.js [hunk 6]
```diff
@@ -3,103 +3,58 @@ import "../assets/css/fontawesome.min.css";
...
+  console.log("9999 cat: ", cat);
   useEffect(() => {
     const fetchSubCategories = async () => {
       try {
         const res = await axios.get(
-          "http://localhost:8803/sub_categories/" + location.state.id
+          "http://localhost:8803/sub_categories/" + cat
         );
         datalist = res.data;
         document.getElementById("h1").innerHTML = res.data[0].sc_name;
```
Comment (wazmaaali) [inline ac_posystem/src/pages/MainCategory.js]: Never assign static values to restrict website, instead make it dynamic and assign values in dynamic way.

**Review by Madhavareddymukku — COMMENTED** [review]
This class might be used for future purposes.

**Comment by Sravaniksheerasagar** [comment]
All previous bugs are identified and fixed

**Pull request merged**